From 36a1d52814562bb589f03226c31ca823f0518fd2 Mon Sep 17 00:00:00 2001
From: Eli Zaretskii <eliz@gnu.org>
Date: Wed, 14 Mar 2018 20:11:16 +0200
Subject: [PATCH] Fix problems caused by fontconfig-2.13.0

* src/xterm.c (x_term_init): Call fixup_locale after
xg_initialize, to countermand the call to setlocale in some
versions of fontconfig.  (Bug#30788)
---
 src/xterm.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/xterm.c b/src/xterm.c
index c5163aa990..7b445e5f46 100644
--- a/src/xterm.c
+++ b/src/xterm.c
@@ -12411,12 +12411,16 @@ x_term_init (Lisp_Object display_name, char *xrm_option, char *resource_name)
         unrequest_sigio (); /* See comment in x_display_ok.  */
         gtk_init (&argc, &argv2);
         request_sigio ();
-        fixup_locale ();
 
         g_log_remove_handler ("GLib", id);
 
         xg_initialize ();
 
+	/* Do this after the call to xg_initialize, because when
+	   Fontconfig is used, xg_initialize calls its initialization
+	   function which in some versions of Fontconfig calls setlocale.  */
+	fixup_locale ();
+
         dpy = DEFAULT_GDK_DISPLAY ();
 
 #if ! GTK_CHECK_VERSION (2, 90, 0)
-- 
2.14.3

